File: src/prod/src/ServiceModel/naming/PagedSubNameInfoList.cpp
// ------------------------------------------------------------
// Copyright (c) Microsoft Corporation.  All rights reserved.
// Licensed under the MIT License (MIT). See License.txt in the repo root for license information.
// ------------------------------------------------------------

#include "stdafx.h"

namespace Naming
{
    using namespace std;
    using namespace Common;

    PagedSubNameInfoList::PagedSubNameInfoList()
        : EnumerateSubNamesResult()
        , continuationTokenString_()
        , isConsistent_(false)
    {
    }

    PagedSubNameInfoList::PagedSubNameInfoList(
        EnumerateSubNamesResult && enumerateSubNamesResult,
        std::wstring && continuationTokenString)
        : EnumerateSubNamesResult(std::move(enumerateSubNamesResult))
        , continuationTokenString_(std::move(continuationTokenString))
    {
        isConsistent_ = false;
        if ((FABRIC_ENUMERATION_STATUS & FABRIC_ENUMERATION_CONSISTENT_MASK) != 0)
        {
            isConsistent_ = true;
        }
    }
}

File: src/prod/test/httpgatewaytest/PagedSubNameInfoList.cs
// ------------------------------------------------------------
// Copyright (c) Microsoft Corporation.  All rights reserved.
// Licensed under the MIT License (MIT). See License.txt in the repo root for license information.
// ------------------------------------------------------------

using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Runtime.Serialization;
using System.Fabric;

namespace System.Fabric.Test.HttpGatewayTest
{
    // Binary and Double values need their own class because they
    // are not sent as strings.

    [Serializable]
    [DataContract]
    class PagedSubNameInfoList
    {

        public PagedSubNameInfoList()
        {
            ContinuationToken = "";
            IsConsistent = false;
            SubNames = new List<Uri>();
        }

        [DataMember]
        public string ContinuationToken
        {
            get;
            set;
        }

        [DataMember]
        public bool IsConsistent
        {
            get;
            set;
        }

        [DataMember]
        public List<Uri> SubNames
        {
            get;
            set;
        }
    }
}